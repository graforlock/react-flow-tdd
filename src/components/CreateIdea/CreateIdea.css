.create-idea__section {
  position: fixed;
  right: 0;
  bottom: 0;
  box-shadow: gainsboro 0 2.5px 7.5px;
  border-top-left-radius: 10px;
  padding: 20px;
}

.create-idea__section--white {
  background: white;
}

.create-idea__section--blue {
  background: #32557E;
}

.create-idea__add {
  color: white;
  cursor: pointer;
  font-weight: bold;
  font-size: 18px;
  border: none;
  border-top-left-radius: 10px;
  background: #32557E;
}