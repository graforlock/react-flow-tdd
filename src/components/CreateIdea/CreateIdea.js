// @flow
import React from 'react'
import { connect } from 'react-redux'

import { fetchCreateIdea, enableEditorWriting } from '../../actions'

import CreateForm from '../CreateForm'
import './CreateIdea.css'

type Props = {
  writingMode: boolean,
  fetchCreateIdea: Function,
  enableEditorWriting: Function
}

const CreateIdea = ({
  fetchCreateIdea,
  writingMode,
  enableEditorWriting,
  resetForm
}: Props) => {
  const handleSubmit = e => {
    fetchCreateIdea(e)
  }
  const blueVariant = writingMode ? '' : 'create-idea__section--blue'
  return (
    <section className={`create-idea__section ${blueVariant}`}>
      {writingMode
        ? <CreateForm onSubmit={handleSubmit} />
        : <button className='create-idea__add'
          onClick={enableEditorWriting}>+</button>}
    </section>
  )
}

const mapStateToProps = ({ editor: { writingMode } }) => {
  return { writingMode }
}

export { CreateIdea }
export default connect(mapStateToProps, {
  fetchCreateIdea,
  enableEditorWriting
})(CreateIdea)
