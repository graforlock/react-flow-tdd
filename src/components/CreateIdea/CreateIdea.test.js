// @flow
import React from 'react'
import type { ShallowWrapper } from 'enzyme'
import { shallow } from 'enzyme'

import { CreateIdea } from './CreateIdea'

describe('CreateIdea', () => {
  let wrapper: ShallowWrapper
  let fetchCreateIdea

  beforeEach(() => {
    fetchCreateIdea = jest.fn()
    wrapper = shallow(
      <CreateIdea enableEditorWriting={jest.fn()}
        fetchCreateIdea={fetchCreateIdea} writingMode={false} />
      )
  })

  it('renders component and matches current default snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders component and matches snapshot when writingMode is true', () => {
    wrapper.setProps({ writingMode: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('contains a working ReduxForm that creates a new idea', () => {
    wrapper.setProps({ writingMode: true })
    wrapper.find('ReduxForm').simulate('submit')
    expect(fetchCreateIdea).toHaveBeenCalled()
  })
})
