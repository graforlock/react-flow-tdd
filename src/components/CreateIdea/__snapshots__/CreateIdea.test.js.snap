// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIdea renders component and matches current default snapshot 1`] = `
<section
  className="create-idea__section create-idea__section--blue"
>
  <button
    className="create-idea__add"
    onClick={[Function]}
  >
    +
  </button>
</section>
`;

exports[`CreateIdea renders component and matches snapshot when writingMode is true 1`] = `
<section
  className="create-idea__section "
>
  <ReduxForm
    onSubmit={[Function]}
  />
</section>
`;
